src/networking.c

```diff
@@ -850,11 +855,16 @@ void AddReplyFromClient(client *dst, client *src) {
     if (prepareClientToWrite(dst) != C_OK)
         return;
     addReplyProto(dst,src->buf, src->bufpos);
+
+    /* we're bypassing _addReplyProtoToList, so we need to add the pre/post
+     * checks in it. */
+    if (dst->flags & CLIENT_CLOSE_AFTER_REPLY) return;
```

Comment:
Hi @oranagra Why we add this check, `prepareClientToWrite` will do such check.

Reply:
i've put it here because it's related to the code below (which corresponds to _addReplyProtoToList), but actually moving the check to the top (or counting on the one that's already in prepareClientToWrite) is not enough.
since the flag can be set from within addReplyProto.
i.e. addReplyProto will abort (not adding src->buf to the reply), and then we'll be unaware of this and add  src->reply to the reply anyway.

i'll fix the comment to describe that.

Reply:
CLIENT_CLOSE_AFTER_REPLY or  CLIENT_CLOSE_ASAP?
can we call `prepareClientToWrite` again?  there may be some other flags we need to check.

Reply:
i'm sorry, i mixed up these flags (this is confusing).
now i think we need to check both.
we're bypassing _addReplyProtoToList, so we need to replicate the check that it has (CLIENT_CLOSE_AFTER_REPLY), but we indeed need to re-call prepareClientToWrite (since CLIENT_CLOSE_AFTER_REPLY might have changed).

@ShooterIT shall i push a fix for that, or do you see other issues?

Reply:
Oh, i also miss somethings, i initially think `prepareClientToWrite` will reject clients with CLIENT_CLOSE_AFTER_REPLY  flag. 

Yes, I think we should check both.

- We shouldn't write anything if `prepareClientToWrite` returns error because of last `addReplyProto`
- listJoin just like _addReplyProtoToList, we should check CLIENT_CLOSE_AFTER_REPLY.